samples/dashboards/sync/custom-component-sync/demo.js

```diff
@@ -0,0 +1,286 @@
+const { Component, ComponentRegistry } = Dashboards;
+
+class AveragesMirror extends Component {
+    constructor(cell, options) {
+        super(cell, options);
+        this.type = 'AveragesMirror';
+        this.createDOMStructure();
+        this.xColumn = [];
+        this.yColumn = [];
+        this.sync = new Component.Sync(
+            this,
+            this.syncHandlers
+        );
+        return this;
+    }
+
+    createDOMStructure() {
+        this.slider = document.createElement('input');
+        this.handleLabel = document.createElement('span');
+        this.leftSide = document.createElement('div');
+        this.rightSide = document.createElement('div');
+
+        const handleLabelContainer = document.createElement('div');
```

Comment:
It's a bit messy and difficult to read- could you please divide it and add comments if needed?